Add example for cross-repo links (#227)
grafana/writers-toolkit: docs/sources/writing-guide/references/index.md

Hugo has built-in shortcodes for creating links to documents.
The `ref` and `relref` shortcodes display the absolute and relative permalinks to a document, respectively.

> >**Note**: When linking to specific versions or across repositories, use standard markdown links. Read the [Versions and cross-pository linking]({{< relref "#versions-and-cross-repository-linking" .}}) section for details.

Author: ```suggestion
>**Note**: When linking to specific versions or across repositories, use standard markdown links. Read the [Versions and cross-pository linking]({{< relref "#versions-and-cross-repository-linking" >}}) section for details.
```

> >**Note**: When linking to specific versions or across repositories, use standard markdown links. Read the [Versions and cross-pository linking]({{< relref "#versions-and-cross-repository-linking" >}}) section for details.

Reviewer: Not sure if you mean for this to say "cross-repository" or not
```suggestion
>**Note**: When linking to specific versions or across repositories, use standard markdown links. Read the [Versions and cross-repository linking]({{< relref "#versions-and-cross-repository-linking" >}}) section for details.
```

| `/docs/sources/branch/other.md`  | `/docs/sources/branch/_index.md` | `{{</* relref "_index.md" */>}}`           |
| `/docs/sources/branch/other.md`  | `/docs/sources/leaf/index.md`    | `{{</* relref "../leaf/index.md" */>}}`    |

## Versions and cross-repository linking

> For Hugo's purposes, you can't address other versions of the docs, such as a version-specific archived docs set (`https://grafana.com/docs/grafana/v8.5/`, etc.) or `/next/` docs for links in content residing in `/latest/`, using Hugo references.

Reviewer 2: ```suggestion
For Hugo's purposes, you can't address other versions of the docs, such as a version-specific archived docs set (`https://grafana.com/docs/grafana/v8.5/` and so forth) or `/next/` docs for links in content residing in `/latest/`, using Hugo references.
```

Reviewer 2: nit :)

Hugo references addressed across different products' docs, such as from `/docs/grafana/` to `/docs/loki/` and vice-versa, as well as references from docs addressed to other Hugo-published content on grafana.com, can also be unpredictably addressed.

To avoid broken links in these situations on grafana.com, use regular Markdown link syntax (`[link text](/docs/repo/version/folder/file/)`) instead of Hugo references. To ensure the links work in local builds, staging environments, and the live website, you **shouldn't** use a fully qualified URL with `https://grafana.com` for links to other content on grafana.com.

> For cross-repository links, use a standard markdown link, like this:

Reviewer: Wondering if before we lay out the example if we can add the structure. Something like: 

```suggestion
For cross-repository links, use a standard markdown link, with the link structured like so:
`\docs\repo\page\`

For example:
```

| `/docs/sources/branch/other.md`  | `/docs/sources/branch/_index.md` | `{{</* relref "_index.md" */>}}`           |
| `/docs/sources/branch/other.md`  | `/docs/sources/leaf/index.md`    | `{{</* relref "../leaf/index.md" */>}}`    |

## Versions and cross-repository linking

> For Hugo's purposes, you can't address other versions of the docs, such as a version-specific archived docs set (`https://grafana.com/docs/grafana/v8.5/` and so forth) or `/next/` docs for links in content residing in `/latest/`, using Hugo references.

Reviewer 3: This isn't a Hugo thing, this is a limitation of our local webserver environments.

With the `make-docs` script, it is now possible to build multiple documentation sets together and I have demonstrated cross project link checking in https://github.com/grafana/tempo/pull/2299.

Author: ```suggestion
For Grafana's webserver environments, you can't address other versions of the docs, such as a version-specific archived docs set (`https://grafana.com/docs/grafana/v8.5/` and so forth) or `/next/` docs for links in content residing in `/latest/`, using Hugo references.
```

> Hugo references addressed across different products' docs, such as from `/docs/grafana/` to `/docs/loki/` and vice-versa, as well as references from docs addressed to other Hugo-published content on grafana.com, can also be unpredictably addressed.

Reviewer 3: What does this mean?
Hugo relrefs require a specific way of specifying the parameter but they shouldn't behave unpredictably.

Author: ```suggestion
Hugo references addressed across different products' docs, such as from `/docs/grafana/` to `/docs/loki/` and vice-versa, as well as references from docs addressed to other Hugo-published content on grafana.com, can also be predictably addressed.
```

To avoid broken links in these situations on grafana.com, use regular Markdown link syntax (`[link text](/docs/repo/version/folder/file/)`) instead of Hugo references. To ensure the links work in local builds, staging environments, and the live website, you **shouldn't** use a fully qualified URL with `https://grafana.com` for links to other content on grafana.com.

For cross-repository links, use a standard markdown link, with the link structured like so:
`\docs\repo\page\`

For example:

```markdown
This is an [example cross-repository link](/docs/grafana/whatsnew) to the Grafana repository.
```

> Using a Hugo `relref` in a cross-repository link or a link to a specific version can result in a page not found error message when running `make docs`.

Reviewer 3: Only if the linked content isn't also mounted when using the `make-docs` script.

> Unlike references, Hugo does _not_ confirm that these link destinations exist during its build, so manually confirm that the published links in a local build and on the published website point correctly.

Reviewer 3: With partial URIs, you also cannot check these links without the content mounted. `/docs/grafana/latest/` from `/docs/tempo/latest` won't resolve unless you have both projects mounted in the webserver.

Author: I will add what you noted :) 
```suggestion
Unlike references, Hugo does _not_ confirm that these link destinations exist during its build, so manually confirm that the published links in a local build and on the published website point correctly.
With partial URIs, you also cannot check these links without the content mounted. For example, `/docs/grafana/latest/ from /docs/tempo/latest` won't resolve unless you have both projects mounted in the webserver.
```

If the destination file or its containing directory has a period (`.`) in the path, you must link to the source file directly.
{{% /admonition %}}

## Versions and cross-repository linking

For Grafana's webserver environments, you can't address other versions of the docs, such as a version-specific archived docs set (`https://grafana.com/docs/grafana/v8.5/` and so forth) or `/next/` docs for links in content residing in `/latest/`, using Hugo references.

Hugo references addressed across different products' docs, such as from `/docs/grafana/` to `/docs/loki/` and vice-versa, as well as references from docs addressed to other Hugo-published content on grafana.com, can also be predictably addressed.

To avoid broken links in these situations on grafana.com, use regular Markdown link syntax (`[link text](/docs/repo/version/folder/file/)`) instead of Hugo references. To ensure the links work in local builds, staging environments, and the live website, you **shouldn't** use a fully qualified URL with `https://grafana.com` for links to other content on grafana.com.

> For cross-repository links, use a standard markdown link, with the link structured like so:
> `\docs\repo\page\`

Author: ```suggestion
For cross-repository links, use a standard markdown link, with the link structured like this: `\docs\repo\page\`.
```

For example:

```markdown
This is an [example cross-repository link](/docs/grafana/whatsnew) to the Grafana repository.
```

Using a Hugo `relref` in a cross-repository link or a link to a specific version can result in a page not found error message when running `make docs` if the linked content isn't mounted when using the script.

Unlike references, Hugo does _not_ confirm that these link destinations exist during its build, so manually confirm that the published links in a local build and on the published website point correctly.

> With partial URIs, you also cannot check these links without the content mounted. For example, `/docs/grafana/latest/ from /docs/tempo/latest` won't resolve unless you have both projects mounted in the webserver.

Reviewer 3: ```suggestion
With partial URIs, you also cannot check these links without the content mounted. For example, `/docs/grafana/latest/` from `/docs/tempo/latest` won't resolve unless you have both projects mounted in the webserver.
```
